packages/perseus/src/__tests__/widgets.test.ts

```diff
@@ -26,6 +26,158 @@ describe("Widget API support", () => {
+        }
+    });
+
+    it("grapher widget getUserInputFromProps should return the correct user input", () => {
+        const Widget = Widgets.getWidget("grapher");
+
+        if (Widget && "getUserInputFromProps" in Widget) {
+            const props = {
+                plot: {
+                    type: "linear",
+                    coords: [
+                        [0, 0],
+                        [1, 1],
+                    ],
+                },
+            };
+
+            // @ts-expect-error - TS2339 - Property 'getUserInputFromProps' does not exist on type 'ComponentType<any>'.
+            const userInput = Widget.getUserInputFromProps(props);
+            expect(userInput).toEqual({
+                type: "linear",
+                coords: [
+                    [0, 0],
+                    [1, 1],
+                ],
+            });
+        } else {
+            throw new Error("Widget does not have getUserInputFromProps");
+        }
+    });
+
+    it("input-number widget getUserInputFromProps should return the correct user input", () => {
+        const Widget = Widgets.getWidget("input-number");
+
+        if (Widget && "getUserInputFromProps" in Widget) {
+            const props = {
+                currentValue: "42",
+            };
+
+            // @ts-expect-error - TS2339 - Property 'getUserInputFromProps' does not exist on type 'ComponentType<any>'.
+            const userInput = Widget.getUserInputFromProps(props);
+            expect(userInput).toEqual({
+                currentValue: "42",
+            });
+        } else {
+            throw new Error("Widget does not have getUserInputFromProps");
+        }
+    });
+
+    it("interactive-graph widget getUserInputFromProps should return the correct user input", () => {
+        const Widget = Widgets.getWidget("interactive-graph");
+
+        if (Widget && "getUserInputFromProps" in Widget) {
+            const props = {
+                graph: {
+                    type: "circle",
+                    radius: 5,
+                    center: [0, 0],
+                },
+            };
+
+            // @ts-expect-error - TS2339 - Property 'getUserInputFromProps' does not exist on type 'ComponentType<any>'.
+            const userInput = Widget.getUserInputFromProps(props);
+            expect(userInput).toEqual({
+                type: "circle",
+                radius: 5,
+                center: [0, 0],
+            });
+        } else {
+            throw new Error("Widget does not have getUserInputFromProps");
+        }
+    });
+
+    it("numeric-input widget getUserInputFromProps should return the correct user input", () => {
+        const Widget = Widgets.getWidget("numeric-input");
+
+        if (Widget && "getUserInputFromProps" in Widget) {
+            const props = {
+                currentValue: 42,
+            };
+
+            // @ts-expect-error - TS2339 - Property 'getUserInputFromProps' does not exist on type 'ComponentType<any>'.
```

Comment:
Ouch, I guess because `getWidget` returns `React.ComponentType<any>`.  No changes needed, just wish we could type `getWidget` better.

Reply:
Yeah 😦